### PR TITLE
Fix Tests for New Operator Version

### PR DESCRIPTION
**Pull request by TimPansino**
* Fix tests for new operator version by switching all test app helm charts to pod selectors.
* Add read only DockerHub token to fix rate limiting.
* NOTE: PHP seems to be working locally, but there's some problem getting the data to report correctly in CI.

### CODE REVIEW

**Comment by TimPansino** [comment]
I think this PR is going to be blocked until the new helm charts are published.

**Review by fallwith — APPROVED** [review]

File: tests/Makefile
Comment (bduranleau-nr) [inline tests/Makefile:92]:
```suggestion
logs-testapp:
```

File: .github/workflows/test.yml
Comment (tippmar-nr) [inline .github/workflows/test.yml:79]: Can we keep this line in place in case folks (like .NET) want to test locally and send data to Staging?

File: tests/dotnet/chart/templates/deployment.yaml
Comment (tippmar-nr) [inline tests/dotnet/chart/templates/deployment.yaml:42]: For unknown reasons, I had to use `LoadBalancer` here to get the .NET tests to work locally. Maybe this works now, I don't know.

**Review by jtduffy — DISMISSED** [review]

File: .github/workflows/test.yml
Comment (TimPansino) [reply .github/workflows/test.yml:79]: How are you running github actions files locally? I could add that line to the makefile for local testing though.

File: tests/dotnet/chart/templates/deployment.yaml
Comment (chynesNR) [inline tests/dotnet/chart/templates/deployment.yaml:28]: I don't think this will work, since we want `"NOTSET"` to be quoted

File: .github/workflows/test.yml
Comment (tippmar-nr) [reply .github/workflows/test.yml:79]: Oh, maybe that's my bad. I wasn't thinking GHA when I left the comment. Locally, we can't use the makefile either (silly .NET) - so I just set environment variables as needed and manually invoke the steps.

In any case, this is non-issue.

File: tests/dotnet/chart/templates/deployment.yaml
Comment (TimPansino) [reply tests/dotnet/chart/templates/deployment.yaml:42]: I checked and there's no issues with NodePort vs LoadBalancer, as long as you add `/WeatherForecast` to the URL that the Makefile tries to open since it doesn't know about paths and tries to open `/`.

File: tests/dotnet/chart/templates/deployment.yaml
Comment (TimPansino) [reply tests/dotnet/chart/templates/deployment.yaml:28]: The double brackets {{ }} are preprocessed by helm, and the quoting on NOTSET is consumed there. The final output from helm templating is this, which I ran by hand.

```
> helm template -s templates/deployment.yaml ./tests/python/chart
...
          - name: NEW_RELIC_LABELS
            value: "testKey:NOTSET"
...
```

Comment (chynesNR) [reply tests/dotnet/chart/templates/deployment.yaml:28]: Ah! Ok, got it

**Comment by chynesNR** [comment]
> As a random side note, anyone know why reviews are being dismissed when that setting isn't even enabled on branch protections?

I believe it's enabled at the org level now?

**Comment by tippmar-nr** [comment]
> As a random side note, anyone know why reviews are being dismissed when that setting isn't even enabled on branch protections?

@TimPansino It's probably coming from the global config settings that were forced on all repos. It's obnoxious on the one hand, especially when there are comments that have to be addressed -- but it's also a reasonable protection against someone slipping in unreviewed changes.

**Review by chynesNR — APPROVED** [review]

**Review by fallwith — APPROVED** [review]

**Review by deleonenriqueta — APPROVED** [review]

File: tests/php/test-specs.yml
Comment (bduranleau-nr) [inline tests/php/test-specs.yml:12]:
```suggestion
      - curl --fail-with-body $(minikube service test-app-php-service --url -n default)
```
I think these sleeps are unnecessary since the test app already `sleeps` and we've identified that timing isn't the problem in these test failures.

**Review by bduranleau-nr — APPROVED** [review]

**Review by jtduffy — APPROVED** [review]

**Review by svetlanabrennan — APPROVED** [review]

File: tests/php/test-specs.yml
Comment (TimPansino) [reply tests/php/test-specs.yml:12]: You're right, I'll open the follow up PR to test PHP changes for when they release the fix and I'll trim those back out. I'd rather not dismiss all our approvals again 😅

**Pull request merged**
